src/Kernel/IQSharpEngine.cs

```diff
@@ -41,20 +58,38 @@ public class IQSharpEngine : BaseEngine
             IServiceProvider services,
             IConfigurationSource configurationSource,
             PerformanceMonitor performanceMonitor,
-            IShellRouter shellRouter,
-            IEventService eventService,
-            IMagicSymbolResolver magicSymbolResolver,
-            IReferences references
+            IShellRouter shellRouter
         ) : base(shell, shellRouter, context, logger, services)
         {
             this.performanceMonitor = performanceMonitor;
             performanceMonitor.Start();
+            this.configurationSource = configurationSource;
+            this.services = services;
+            this.logger = logger;
+        }
+
+        /// <inheritdoc />
+        public override void Start()
+        {
+            base.Start();
 
+            // Before anything else, make sure to start the right background
+            // thread on the Q# compilation loader to initialize serializers
+            // and deserializers. Since that runs in the background, starting
+            // the engine should not be blocked, and other services can
+            // continue to initialize while the Q# compilation loader works.
```

Comment:
If there is a GH issue, perhaps link it here to explain what problem this is solving

Reply:
I don't think there's an issue that was created for this already, so much as that @cesarzc and I noticed this while walking through the code yesterday to understand the initialization performance better. Perhaps linking to the PR that introduced the serializer initialization methods could be useful, though? Let me go on and do that, then.

Reply:
Here's the PR that introduced serializer/deserializer initialization methos: https://github.com/microsoft/qsharp-compiler/pull/727

Reply:
Thanks both!